test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -709,6 +711,26 @@ foo``u{2195}abc
             param($Script)
              {[ScriptBlock]::Create($Script).Invoke()} | Should -Throw
         }
+
+        $testLiteralTypes = @(
+            @{ Script = "10u"; Expected = "UInt32" }
+            @{ Script = "10ul"; Expected = "UInt64" }
+            @{ Script = "10us"; Expected = "UInt16" }
+            @{ Script = "10s"; Expected = "Int16" }
+            @{ Script = "10l"; Expected = "Int64" }
+            @{ Script = "10d"; Expected = "Decimal" }
+            @{ Script = "0x10u"; Expected = "UInt32" }
+            @{ Script = "0x10ul"; Expected = "UInt64" }
+            @{ Script = "0x10us"; Expected = "UInt16" }
+            @{ Script = "0x10s"; Expected = "Int16" }
+            @{ Script = "0x10l"; Expected = "Int64" }
+            # No decimal hex literals (d is a valid hex char, and decimal parsing does not accept hex values)
+            @{ Script = "0x10d"; Expected = "Int32" }
+        )
+        It "<Script> should result in an [<Expected>] value" -TestCases $testLiteralTypes {
+            param($Script, $Expected)
+            ExecuteCommand $Script | Should -BeOfType $Expected
```

Comment:
I think it should be added to tests where we check values.

Reply:
Hmm... Oh, you mean with those others there like the long checks...

I know what you mean, but after running into the quick with Int16 values I felt it perhaps might be a little more prudent to directly check types rather than checking the way some of the other tests 'check type'. It's hard to guarantee the value won't just be cast into the same form when it may not be quite correct.

Reply:
We can test type before any operations on value.

Reply:
I'm not following, I think. The existing test for values doesn't consider type except in a few cases, and even in those the type checks are implicit, not explicit, and casting may occur.

Reply:
Can we do:
```powershell
	Context "Numerical Notations Tests (starting at line 2374 to line 2452)" {
        $testData = @(
			#Test various numbers using the standard notation.
            @{ Script = "0"; Expected = "0" }
			@{ Script = "-2"; Expected = "-2"; ExpectedType = [int32] }
			@{ Script = "2"; Expected = "2"; ExpectedType = [int32] }
			@{ Script = $([int32]::MaxValue); Expected = $([int32]::MaxValue); ExpectedType = [int32] }
			@{ Script = $([int32]::MinValue); Expected = $([int32]::MinValue); ExpectedType = [int32] }

...

        It "<Script> should return <Expected>" -TestCases $testData {
            param ( $Script, $Expected, $ExpectedType )
            $res = ExecuteCommand $Script
            $res | Should -Be $Expected
             $res | Should -BeType $ExpectedType
       }

Reply:
Roll it all into one.... Hm. Yeap, should be able to. I'll check into it shortly.

Reply:
If we continue in follow PR to add optinizations in the code and fix a hex sign it is important check not only parsing each string here but also test a result type. I think this will help a lot in follow PR avoid regressions. 
If you discover that we check here not all numeric formats please add new lines in the test.